lms/product/canvas/_plugin/grading.py

```diff
@@ -0,0 +1,32 @@
+class CanvasGradingPlugin:
+    @staticmethod
+    def configure_grading_for_launch(request, js_config, assignment):
+        """
+        Configure grading during a launch.
+
+        In canvas we use Speed Grader, we configure the frontend to use it.
+        """
+        lti_params = request.lti_params
+        lti_user = request.lti_user
+
+        # For students in Canvas with grades to submit we need to enable
+        # Speedgrader settings for gradable assignments
+        # `lis_result_sourcedid` associates a specific user with an
+        # assignment.
+        if (
+            assignment.is_gradable
+            and lti_user.is_learner
+            and lti_params.get("lis_result_sourcedid")
+        ):
+            js_config.add_canvas_speedgrader_settings(assignment.document_url)
```

Comment:
Should these settings be here? If this is Canvas specific, and this is the place that calls it, seems like the time?